manager/assets/modext/widgets/resource/modx.panel.trash.js

```diff
@@ -0,0 +1,442 @@
+            ,text: _('trash.purge_all')
+            ,id: 'modx-purge-all'
+            ,cls: 'x-form-purge-all red'
+            ,listeners: {
+                'click': {fn: this.purgeAll,scope: this}
+            }
+        },{
+            xtype: 'button'
+            ,text: _('trash.restore_all')
+            ,id: 'modx-restore-all'
+            ,cls: 'x-form-restore-all green'
+            ,listeners: {
+                'click': {fn: this.restoreAll,scope: this}
+            }
+        },'->',{
+            xtype: 'textfield'
+            ,name: 'search'
+            ,id: 'modx-source-search'
+            ,cls: 'x-form-filter'
+            ,emptyText: _('search_ellipsis')
+            ,listeners: {
+                'change': {fn: this.search,scope: this}
+                ,'render': {
+                    fn: function (cmp) {
+                        new Ext.KeyMap(cmp.getEl(),{
+                            key: Ext.EventObject.ENTER
+                            ,fn: this.blur
+                            ,scope: cmp
+                        });
+                    },scope: this
+                }
+            }
+        },{
+            xtype: 'button'
+            ,text: _('filter_clear')
+            ,id: 'modx-filter-clear'
+            ,cls: 'x-form-filter-clear'
+            ,listeners: {
+                'click': {fn: this.clearFilter,scope: this}
+            }
+        }]
+    });
+    MODx.grid.Trash.superclass.constructor.call(this,config);
+};
+
+Ext.extend(MODx.grid.Trash,MODx.grid.Grid,{
+    getMenu: function () {
+        var r = this.getSelectionModel().getSelected();
+        var p = r.data.cls;
+
+        var m = [];
+        if (this.getSelectionModel().getCount() > 1) {
+            m.push({
+                text: _('trash.selected_purge')
+                ,handler: this.purgeSelected
+                ,scope: this
+            });
+            m.push({
+                text: _('trash.selected_restore')
+                ,handler: this.restoreSelected
+                ,scope: this
+            });
+        } else {
+            if (p.indexOf('trashpurge') != -1) {
+                m.push({
+                    text: _('trash.purge')
+                    ,handler: this.purgeResource
+                });
+            }
+            if (p.indexOf('trashundelete') != -1) {
+                m.push({
+                    text: _('trash.restore')
+                    ,handler: this.restoreResource
+                });
+            }
+        }
+        if (m.length > 0) {
+            this.addContextMenuItem(m);
+        }
+    }
+    ,purgeResource: function () {
+        MODx.msg.confirm({
+            title: _('trash.purge_confirm_title')
+            ,text: _('trash.purge_confirm_message',{
+                'list': this.listResources('<br/>')
+            })
+            ,url: this.config.url
+            ,params: {
+                action: 'resource/trash/purge'
+                ,id: this.menu.record.id
```

Comment:
Purge processor expectes `ids` variable, so here will be an error if I`m trying to remove only one resource with context menu
![screen shot 2018-04-06 at 14 02 55](https://user-images.githubusercontent.com/1257284/38420559-6e233b5c-39a4-11e8-9a19-d104042b6c36.png)

Changing `id` to `ids` solves problem.